cmd/protoc-gen-go-aip/internal/genaip/testdata/test/multipattern/testdata_aip.go

```diff
@@ -82,12 +82,19 @@ func (n ShelvesBookResourceName) MarshalString() (string, error) {
 }
 
 func (n *ShelvesBookResourceName) UnmarshalString(name string) error {
-	return resourcename.Sscan(
+	err := resourcename.Sscan(
 		name,
 		"shelves/{shelf}/books/{book}",
 		&n.Shelf,
 		&n.Book,
 	)
+	if err != nil {
+		return err
+	}
+	if err := n.Validate(); err != nil {
```

Comment:
You could return the error directly.
`return n.Validate()`

Reply:
fixed